File: lecture_010(RecursionwithArraylists)/l_007_printkeypadcombination.java
import java.util.Scanner;
public class l_007_printkeypadcombination {
    public static Scanner scn = new Scanner(System.in);
    public static void main(String[] args) {
        String str = scn.nextLine();
        printKpc(str,"");
    }
    static String[] codes = {".;", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tu", "vwx", "yz"};
    public static void printKpc(String ques, String ans) {
        if(ques.length()==0) {
            System.out.println(ans);
            return;
        }
        char ch = ques.charAt(0);
        String roq = ques.substring(1);

        String deep = codes[ch-'0'];
        for(int i=0; i<deep.length(); i++) {
            char option = deep.charAt(i);
            printKpc(roq,ans+option);
        }
    }
}

File: lecture_010(RecursionwithArraylists)/l_007_printkeypadcombination.cpp
#include<iostream>
#include<vector>
#include<string>
using namespace std;

vector<string> codes = {".;", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tu", "vwx", "yz"};
void printkpc(string ques, string ans) {
    if(ques.length()==0) {
        cout<<ans<<endl;
        return;
    }
    char ch = ques[0];
    string roq = ques.substr(1);

    string deep = codes[ch-'0'];
    for(int i=0; i<deep.length(); i++) {
        char option = deep[i];
        printkpc(roq,ans+option);
    }
}
int main() {
    string str;
    getline(cin,str);
    printkpc(str,"");

    return 0;
}